fix(migrate)!: app router and next-intl

GMSoudersJr opened the pull request:
I keep getting what looks like the same error for all pages except home, but the messages do not help. There is one more thing I will try but I wanted to get some advice before I spend more time fixing this.

BREAKING CHANGE

GMSoudersJr committed:
fix(migrate)!: app router and next-intl

I keep getting what looks like the same error for all pages except home,
but the messages do not help. There is one more thing I will trym but
I wanted to get some advice before I spend more time fixing this.

BREAKING CHANGE

GMSoudersJr commented:
I have been going over this for days, but am stuck. I have tried adding a linter for hooks, comparing versions of react and react-dom, moving everything into the app router, I don't know else to do. It is probably something simple that I am not seeing. Yay React!

ivanms1 commented:
thanks for taking the time to work on this, I'll take a look today

GMSoudersJr commented:
No rush at all. One plus of working on this is that it got me back into running. I had to take time to clear my head after some of this.

ivanms1 commented:
I think the problem is the NextSeo component, after removing it the error went away.
Probably not compatible with app router, yay
On the other hand you could use the now built in [metada api](https://nextjs.org/docs/app/building-your-application/optimizing/metadata)
hopefully that helps

GMSoudersJr commented:
Okay, that figures. I did notice that the title wasn't visible in the browser tab. New inspiration!

GMSoudersJr committed:
fix(migration): app routing and seo

Remove next-seo and utilize built-in metadata from next. The errors are
gone and it builds successfully. Metadata is present and changes on
navigation to different pages. Open Graph testing needs to be done to
make sure the shared url looks pretty.

GMSoudersJr commented:
I apologize for how large this is.

Your suggestion worked like a charm. Those errors are gone and it builds successfully.

I implemented the built-in metadata from next.js. The site seems to have the same functionality it did before. Sharing the link needs to be tested, though.

ivanms1 commented on .eslintrc:
could you revert this?
I think this plugin is already included in the next plugin

ivanms1 commented on .eslintrc:
these also can go I think

ivanms1 commented on src/app/[locale]/components/Button/Button.tsx:
is it possible to move the `components` folder outside of the `app` folder.

ivanms1 commented on src/app/[locale]/components/Layout/Layout.tsx:
please remove

ivanms1 commented on src/app/[locale]/components/Layout/Navbar/Navbar.tsx:
always prefer absolute imports

ivanms1 commented on src/app/[locale]/components/Layout/Navbar/Navbar.tsx:
```suggestion
    push(`${LOCALES.ko.code}/${pathname}/`, { locale: LOCALES.ko.code });
```

ivanms1 commented on src/app/[locale]/components/MemberCard/MemberCard.tsx:
remove please

ivanms1 commented on src/app/[locale]/events/Events.tsx:
this was here from before, please remove if not needed

ivanms1 commented on src/app/[locale]/events/page.tsx:
what a name 😆

ivanms1 commented on src/app/[locale]/layout.tsx:
are these comments from the docs?

ivanms1 commented on src/app/layout.tsx:
we could move `src/app/[locale]/components/Layout/Layout.tsx` here, but keep the Navbar and Footer in the components folder.

ivanms1 commented on src/i18n.ts:
maybe we could move this to `src/const/index.ts`
```suggestion
const LOCALES = ["en", "ko"];
```

ivanms1 commented on src/services/members/index.ts:
could you rename this to `REVALIDATE_TIME`?

ivanms1 commented on src/services/meetup/index.ts:
`{ cache: "no-store" }`
this makes it server side rendered?

ivanms1 reviewed (commented):
Great job!
thank you for the hard work, left some minor comments.

GMSoudersJr commented on src/app/[locale]/layout.tsx:
They are. I left them for the next person who contributes. I'll remove them, though.

GMSoudersJr commented on src/services/meetup/index.ts:
Correct. This is the app router way of calling getServerSideProps by using options in fetch. [Docs](https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration#step-6-migrating-data-fetching-methods)

GMSoudersJr commented on .eslintrc:
should I `yarn remove <package-name>` since I added it with yarn? Or simply delete it from the .eslintrc file?

GMSoudersJr commented on src/app/layout.tsx:
🤔 hmmm... Okay. Hadn't crossed my mind.

ivanms1 commented on .eslintrc:
both please 🙇

ivanms1 commented on src/services/meetup/index.ts:
neat 👍

GMSoudersJr commented on src/app/layout.tsx:
I am a little confused by this.

I used this structure to catch errors (create an error boundary) because I was tired of getting those ugly error message windows when I could not get the locale to work. [next-intl docs](https://next-intl-docs.vercel.app/docs/environments/error-files).

Are you suggesting to move the `src/app/[locale]/compnents/Layout/Layout.tsx` file to `src/app` directory?
Would it not be better to place it in `src/app/[locale]/` directory since it is the layout that is used with a locale.

GMSoudersJr committed:
fix(migration): app router suggestions

Remove eslint-plugin-react-hooks with yarn remove and delete from
.eslintrc. Remove commented lines from multiple files. Move components
directory out of app directory. Rename a few variables and
constants. Use absolute imports.

ivanms1 commented on src/app/layout.tsx:
I see what you mean now, this layout is not doing much but it seems that is required because the build fails when it's missing.
We can just leave it as it is, only one thing, could you move BaseLayout and its modules to the components folder?

ivanms1 commented:
looking good, just one more thing, I think we can uninstall next-seo

GMSoudersJr committed:
fix(migrate): move layout to components dir

Move the base layout to components folder. Next-seo was already not
a dependency for some reason.

GMSoudersJr commented:
it's weird. next-seo was not in my package.json

ivanms1 commented:
you removed it already 😄

ivanms1 commented on src/components/MemberCard/MemberCard.tsx:
last one!

ivanms1 commented on tsconfig.json:
could you tell me why was this needed?

ivanms1 commented on src/services/members/index.ts:
I don't think you need to await this

ivanms1 reviewed (commented):
last few comments
almost there!

GMSoudersJr commented on src/services/members/index.ts:
Correct

GMSoudersJr commented on tsconfig.json:
Honestly, I do not remember adding that. Maybe it was added during one of my yarn commands.

I looked into it more, and found that I built the app before pushing my commits. `yarn build` gave this message:
![image](https://github.com/Free-Code-Camp-Seoul/fcc-seoul-website/assets/44438849/39fe1161-d113-48d3-9f00-403ae97f8dde)

GMSoudersJr commented on tsconfig.json:
I guess this is not a best practice since it is optimizing for production and not what we use for development.

GMSoudersJr committed:
fix(migrate): remove comments and await

Remove commented out React import and unncessary await. Possible issue
with building the app before pushing commits. Let me know.

ivanms1 commented on tsconfig.json:
oh that's interesting, I guess we can keep it then

ivanms1 reviewed (approved):
LGTM 🎉 
Thank you for taking the time to do this.

GMSoudersJr commented:
Thanks for taking the time to review it.
